plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -55,13 +55,31 @@ public function setup() {
      * Database structure: on update
      */
     public function structure() {
+        $colBanType = Gdn::structure()->get('Ban')->columns('BanType');
+        $BanTypes = $colBanType->Enum;
+
+        if (!in_array('Fingerprint', $BanTypes)) {
+            $BanTypes[] = 'Fingerprint';
+
+            Gdn::structure()
+                ->table('Ban')
+                ->column('BanType', $BanTypes, false, 'unique')
+                ->set();
+        }
+
         Gdn::structure()
             ->table('User')
             ->column('Troll', 'int', '0')
             ->column('Fingerprint', 'varchar(50)', null, 'index')
             ->set();
     }
 
+    /**
+     * On plugin disable.
+     */
+    public function onDisable() {
+    }
```

Comment:
🤔

Reply:
Remnants of me wanting to cleanup after disabling the plugin. Removed.